Content.Shared/Weather/WeatherComponent.cs

```diff
@@ -29,19 +28,19 @@ public sealed partial class WeatherData
     /// <summary>
     /// When the weather started if relevant.
     /// </summary>
-    [ViewVariables, DataField("startTime", customTypeSerializer: typeof(TimeOffsetSerializer))]
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer))] //TODO: Remove Custom serializer
     public TimeSpan StartTime = TimeSpan.Zero;
 
     /// <summary>
     /// When the applied weather will end.
     /// </summary>
-    [ViewVariables(VVAccess.ReadWrite), DataField("endTime", customTypeSerializer: typeof(TimeOffsetSerializer))]
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer))] //TODO: Remove Custom serializer
```

Comment:
```suggestion
    [DataField, AutoPausedField]
```
You'll also need to add `[AutoGenerateComponentPause]` to the component.

The other one I'm not so sure about. Moving the start time forward while the game is paused doesn't really make sense, but if you don't, then `Duration` is going to be inaccurate. 

I'd probably not add auto-pause to `StartTime` and modify `Duration` to account for paused time...

Reply:
I generally want to edit the weather system, fixing some of its problems with the next PR. Let me just add a pause on startTime now, but fix it with the next PR?

Reply:
i dont think AutoPausedField adds the type serializer for you

Reply:
You can manually handle unpause, iterate through the entries and add to them all. 

Something like this: https://github.com/space-wizards/space-station-14/blob/master/Content.Shared%2FTiming%2FUseDelaySystem.cs#L53

Reply:
> i dont think AutoPausedField adds the type serializer for you

I'm pretty sure TimeSpan is serialized automatically though.

Reply:
its automatically serialized but TimeOffsetSerializer is for round time

Reply:
Huh, til